commit/merkleroot/observation.go

```diff
@@ -318,22 +291,155 @@ type Observer interface {
 	ObserveOffRampNextSeqNums(ctx context.Context) []plugintypes.SeqNumChain
 
 	// ObserveLatestOnRampSeqNums observes the latest OnRamp sequence numbers for each configured source chain.
-	// NOTE: Make sure that caller supports the provided destination chain.
-	ObserveLatestOnRampSeqNums(ctx context.Context, destChain cciptypes.ChainSelector) []plugintypes.SeqNumChain
+	// NOTE: Make sure that caller supports the destination chain.
+	ObserveLatestOnRampSeqNums(ctx context.Context) []plugintypes.SeqNumChain
 
 	// ObserveMerkleRoots computes and returns the merkle roots for the provided sequence number ranges.
 	// NOTE: Make sure that caller supports the provided chains.
 	ObserveMerkleRoots(ctx context.Context, ranges []plugintypes.ChainRange) []cciptypes.MerkleRootChain
 
 	// ObserveRMNRemoteCfg observes the RMN remote config for the given destination chain.
 	// Check implementation specific details to learn if external calls are made, if values are cached, etc...
-	// NOTE: Make sure that caller supports the provided destination chain.
-	ObserveRMNRemoteCfg(ctx context.Context, dstChain cciptypes.ChainSelector) rmntypes.RemoteConfig
+	// NOTE: Make sure that caller supports the destination chain.
+	ObserveRMNRemoteCfg(ctx context.Context) rmntypes.RemoteConfig
 
 	// ObserveFChain observes the FChain for each supported chain. Check implementation specific details to learn
 	// if external calls are made, if values are cached, etc...
 	// NOTE: You can assume that every oracle can call this method, since data are fetched from home chain.
 	ObserveFChain(ctx context.Context) map[cciptypes.ChainSelector]int
+
+	// Close closes the observer and releases any resources.
+	Close() error
+}
+
+// asyncObserver is an Observer implementation that fetches the data asynchronously.
+type asyncObserver struct {
+	lggr                logger.Logger
+	syncObserver        observerImpl
+	cf                  context.CancelFunc
+	mu                  *sync.RWMutex
+	offRampNextSeqNums  []plugintypes.SeqNumChain
+	onRampLatestSeqNums []plugintypes.SeqNumChain
+}
+
+// newAsyncObserver creates a new asyncObserver.
+// It fetches the data from the base observer asynchronously and caches the results.
+// It fetches the data every tickDur and uses a timeout of syncTimeout to kill long RPC calls.
+func newAsyncObserver(lggr logger.Logger, observer observerImpl, tickDur, syncTimeout time.Duration) *asyncObserver {
```

Comment:
Feels like there's quite a bit of components right now that do essentially the same thing - they do some I/O in a background thread and cache state. We already have this for CCIPHome, RMNHome, RMNRemote (I think), and now this component.

Having duplication here is pretty bad because we could have the same bug 3x times instead of just once.

Just trying to see the forest from the trees - feels like we need just a standard `Cache` implementation that is generic enough to handle this (maybe the `State` is just passed in and made generic). I think we had something like this for <= v1.5.

cc @mateusz-sekara

Reply:
Yeh all the above I mentioned are self-syncing as well, just sync periods are slightly different